### DISCUSSION TITLE
Minimum Kubernetes Version: v1.16?

### DISCUSSION THREAD

**Discussion by olix0r** [linkerd/linkerd2#5688] [General] [answered]
Hi folks,

As we prepare stable-2.10, we notice that https://github.com/linkerd/linkerd2/pull/5603, which updates our CRDs to work well with recent k8s versions (v1.19+), is incompatible with Kubernetes v1.15 and earlier. As such, I'm inclined to increase our minimum supported k8s version from 1.13 to 1.16.

K8s v1.15 was released in mid-2019 and has been [unsupported](https://kubernetes.io/docs/setup/release/version-skew-policy/) since the release of v1.18 (Q1 2020).

Does anyone have concerns about updating the minimum version to v1.16 in stable-2.10? Are there any technical reasons we should hold off on updating?

**Comment by mavrick** [comment, marked as answer]
No concerns here

**Comment by christianhuening** [comment]
well, thoroughly late to the party, but we effectively would need K8s 1.15 support :) Is it possible to build a `pre-1-16-crd.yaml` or would that break anything?

**Reply by alpeb** [reply]
If I'm not mistaking, it'd be fine for you to move back to `v1beta1` resources, effectively undoing #5603.

**Reply by christianhuening** [reply]
I guess we don't want to completely undo #5603 but just the version flags. Correct?

**Reply by alpeb** [reply]
Yep. Let me know how it goes!

**Reply by alpeb** [reply]
So I gave this a try, and ended up not being as simple as I thought. The CRDs would require more changes than just the version downgrading, and there might be some go code changes involved as well. Instead, stay tuned for 2.9.5 with a backport for #5722 :wink:

**Reply by christianhuening** [reply]
Aww, I just have the stuff ready to try myself. Sorry to hear it's not working. Ok will stay tuned for 2.9.5 👍 
Can we help somehow?
